packages/outstatic/src/utils/useFileQuery.ts

```diff
@@ -0,0 +1,30 @@
+import { useContext } from 'react'
+import { OutstaticContext } from '../context'
+import { useDocumentQuery } from '../graphql/generated'
+import { useOstSession } from './auth/hooks'
+
+type useFileQueryProps = {
+  file: string
+  skip?: boolean
+}
+
+const useFileQuery = ({ file, skip = false }: useFileQueryProps) => {
```

Comment:
I like this, and will probably reuse this hook in the metadata project when dealing with a single file operation. The corresponding graphql object should probably include `oid` since that's the string we base all our commit logic on.